### ISSUE TITLE
Unable to install spacy>=2.3.3 on Debian 10 in Python 2

### ISSUE THREAD

**Issue by convoliution** [explosion/spaCy#6454]
**tl;dr** I think there's something going on the build requirement specifiers in `spacy>=2.3.3` because from what I can tell, spaCy has an [installation requirement of `blis>=0.4.0,<0.5.0; python_version < "3.6"`](https://github.com/explosion/spaCy/blob/cdca44a/setup.cfg#L50), but the build requirement was apparently changed to `blis<0.8.0,>=0.4.0`.

The issue as I experienced it was that:

1. `spacy==2.3.4` doesn't have a wheel on PyPI for Python 2.7
1. and specifies `blis<0.8.0,>=0.4.0` as a build requirement.
1. This fetches `blis==0.7.3` at build time, which _also_ doesn't provide a wheel for Python 2.7,
1. and itself fails to build on Debian 10 (https://github.com/explosion/cython-blis/issues/42).

But I am able to install `spacy==2.3.2`, and it looks to me like this older version actually specifies `blis<0.5.0,>=0.4.0` to build, and `blis==0.4.1` is able to install without a hitch.

## How to reproduce the problem

1. Run [python:2.7-slim](https://hub.docker.com/layers/python/library/python/2.7-slim/images/sha256-d427179d0d8aa71c0ce958fce5275623fbf59f1af7ee3ff7a7078aefcbf65dd6?context=explore) (i.e. Python 2.7 on Debian 10)
1. `pip install spacy>=2.3.3`

```bash
Collecting spacy>=2.3.3
  Downloading spacy-2.3.4.tar.gz (5.8 MB)
  Installing build dependencies: started
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python /usr/local/lib/python2.7/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-zbRKLh/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools 'cython>=0.25' 'cymem>=2.0.2,<2.1.0' 'preshed>=3.0.2,<3.1.0' 'murmurhash>=0.28.0,<1.1.0' 'thinc>=7.4.1,<7.5.0' 'numpy==1.15.0; python_version<='"'"'3.7'"'"'' 'numpy==1.17.3; python_version=='"'"'3.8'"'"'' 'numpy==1.19.3; python_version=='"'"'3.9'"'"'' 'numpy; python_version>='"'"'3.10'"'"''
       cwd: None
Complete output (475 lines):
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Ignoring numpy: markers 'python_version == "3.8"' don't match your environment
Ignoring numpy: markers 'python_version == "3.9"' don't match your environment
Ignoring numpy: markers 'python_version >= "3.10"' don't match your environment
. . . (output truncated by @convoliution)
Collecting blis<0.8.0,>=0.4.0
  Downloading blis-0.7.3.tar.gz (2.6 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
. . . (output truncated by @convoliution)
Building wheel for blis (PEP 517): started
Building wheel for blis (PEP 517): still running...
Building wheel for blis (PEP 517): still running...
Building wheel for blis (PEP 517): finished with status 'error'
ERROR: Command errored out with exit status 1:
 command: /usr/local/bin/python /usr/local/lib/python2.7/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmp_CaMNp
     cwd: /tmp/pip-install-93_mls/blis
Complete output (365 lines):
('BLIS_COMPILER?', 'None')
running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/blis
copying blis/__init__.py -> build/lib.linux-x86_64-2.7/blis
copying blis/about.py -> build/lib.linux-x86_64-2.7/blis
copying blis/benchmark.py -> build/lib.linux-x86_64-2.7/blis
creating build/lib.linux-x86_64-2.7/blis/tests
copying blis/tests/__init__.py -> build/lib.linux-x86_64-2.7/blis/tests
copying blis/tests/common.py -> build/lib.linux-x86_64-2.7/blis/tests
copying blis/tests/test_dotv.py -> build/lib.linux-x86_64-2.7/blis/tests
copying blis/tests/test_gemm.py -> build/lib.linux-x86_64-2.7/blis/tests
copying blis/cy.pyx -> build/lib.linux-x86_64-2.7/blis
copying blis/py.pyx -> build/lib.linux-x86_64-2.7/blis
copying blis/__init__.pxd -> build/lib.linux-x86_64-2.7/blis
copying blis/cy.pxd -> build/lib.linux-x86_64-2.7/blis
running build_ext
/tmp/pip-build-env-iXXxWa/overlay/lib/python2.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-93_mls/blis/blis/cy.pxd
  tree = Parsing.p_module(s, pxd, full_module_name)
/tmp/pip-build-env-iXXxWa/overlay/lib/python2.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-93_mls/blis/blis/py.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
Processing blis/cy.pyx
Processing blis/py.pyx
('Build options', 'linux2', 'unix')
('[COMMAND]', u'gcc -c /tmp/pip-install-93_mls/blis/blis/_src/config/bulldozer/bli_cntx_init_bulldozer.c -o /tmp/tmpuDdHCe/bli_cntx_init_bulldozer.o -O3 -fPIC -std=c99 -fvisibility=hidden -D_POSIX_C_SOURCE=200112L -DBLIS_VERSION_STRING="0.7.0" -DBLIS_IS_BUILDING_LIBRARY -Iinclude/linux-x86_64 -I./frame/3/ -I./frame/ind/ukernels/ -I./frame/3/ -I./frame/1m/ -I./frame/1f/ -I./frame/1/ -I./frame/include -I/tmp/pip-install-93_mls/blis/blis/_src/include/linux-x86_64')
. . . (output truncated by @convoliution)
```

## Your Environment

* Operating System: Debian 10
* Python Version Used: Python 2.7.18, pip 20.0.2
* spaCy Version Used: >=2.3.3
* Environment Information: python:2.7-slim DockerHub image

**Comment by adrianeboyd** [comment]
Argh, it's been hard to get all the pip and setuptools settings correct here given that we're not building python 2.7 or python 3.5 wheels for new releases. In the vast majority of cases, users don't want to be compiling `blis` because it requires some extra settings.

spacy v2.3.4 is still compatible with blis v0.4.1 where there still are binary wheels for python 2.7, but you'll need to compile `srsly`, `thinc`, and `spacy`. Try this to prefer binary dependencies even if they are older:

```
pip install spacy==2.3.4 --prefer-binary
```

I think you're right that `blis` should be in `setup_requires` and `pyproject.toml` with same `python_version` settings as in `thinc` for this to work as I intended it to, that `blis` nearly always finds a binary package for users running a straightforward `pip install spacy`.
